src/smt/theory_str_noodler/state_len.h

```diff
@@ -90,23 +90,11 @@ namespace smt::noodler {
         }
     };
 
-    /**
-     * @brief Abstract decision procedure. Defines interface for decision 
-     * procedures to be used within z3.
-     */
-    class AbstractDecisionProcedure {
-    public:
-        virtual void initialize(const Instance& inst) =0;
-        virtual bool get_another_solution(const Instance& inst, LengthConstr& out) =0;
-
-        virtual ~AbstractDecisionProcedure() {}
-    };
-
     /**
      * @brief Debug instance of the Decision procedure. Always says SAT and return some length 
      * constraints. Simulates the situation when each instance has exactly 10 noodles.
      */
-    class DecisionProcedureDebug : public AbstractDecisionProcedure {
+    class DecisionProcedureDebug {
```

Comment:
Sorry, I should have been more specific. The debug decision procedure should be moved to `decision_procedure.h` as well and still inherit from `AbstractDecisionProcedure`. The uses of `AbstractDecisionProcedure` should have been kept, you would have to include the `decision_procedure.h` however`.